assets/app.js

```diff
@@ -335,9 +358,25 @@ new Vue({
                 return []
             }
             if(this.rawBaseproducts==null){
-                api.app = this
-                api.action = 'get_baseproducts'
-                api.call()
+                _this = this
```

Comment:
As we just talked about: create an function `fetchResource(type)` which does the fetching of BaseProducts/ Components/Materials/ Metatags so we can avoid duplicate code.